File: app/src/main/java/life/plank/juna/zone/data/network/model/MatchEvent.java
package life.plank.juna.zone.data.network.model;

import lombok.Data;

@Data
public class MatchEvent {
    private Boolean isHomeTeam;
    private String eventType;
    private String playerName;
    private String relatedPlayerName;
    private Integer minute;
    private Integer extraMinute;
    private Boolean injured;
    private String reason;
    private String result;
    private LiveTimeStatus liveTimeStatus;

    public MatchEvent(String eventType, boolean isHomeTeam) {
        this.eventType = eventType;
        this.isHomeTeam = isHomeTeam;
    }

    /**
     * Constructor to add whistle events (kick-off, half-time, full-time etc.) in football timeline.
     */
    public MatchEvent(LiveTimeStatus liveTimeStatus) {
        this.liveTimeStatus = liveTimeStatus;
        this.minute = liveTimeStatus.getMinute();
        this.extraMinute = liveTimeStatus.getExtraMinute();
    }
}

File: app/src/main/java/life/plank/juna/zone/data/model/MatchEvent.kt
package life.plank.juna.zone.data.model

import android.os.Parcelable
import kotlinx.android.parcel.Parcelize
import life.plank.juna.zone.util.common.AppConstants.WHISTLE_EVENT

@Parcelize
data class MatchEvent(
        var isHomeTeam: Boolean,
        var eventType: String,
        var playerName: String = "",
        var relatedPlayerName: String? = "",
        var minute: Int = 0,
        var extraMinute: Int = 0,
        var injured: Boolean? = false,
        var reason: String? = "",
        var result: String? = "",
        var liveTimeStatus: LiveTimeStatus?
) : Parcelable {
    constructor(isHomeTeam: Boolean, eventType: String) : this(isHomeTeam, eventType, "", "", 0, 0, false, "", "", null)

    constructor(liveTimeStatus: LiveTimeStatus) : this(false, WHISTLE_EVENT, "", "", liveTimeStatus.minute, liveTimeStatus.extraMinute, false, "", "", liveTimeStatus)
}